### PR TITLE
Adds new API routes for all ITELL API interactions. Adds API key to the env file.

### PR DESCRIPTION
**Pull request by kyleburgess2025**
As a part of protecting the ITell API, we are implementing required API keys for all ITell API interactions. Instead of calling the API directly, this PR creates four new API routes that will call the ITell API with the deployment API key. 

Some details:
- The new routes are created in the `/app/api` folder. This is because the `pages` router does not support streams; the `app/api` routes are able to create and pass on streams. This has been tested and works on my computer; please test the following to ensure if works on your computer, as well:
- [ ] Chat with ITELL API
- [ ] Grade a summary
- [ ] Grade the answer to a non-summary question
- [ ] Ask for clarification on the grade received for a non-summary question
- Please let me know if I am missing any API calls; I replaced all places where the env variable NEXT_PUBLIC_API_URL was used.
- This also creates a new required env variable; please message me privately to be given a unique developer API key.

### CODE REVIEW

**Review by qiushiyan — CHANGES_REQUESTED** [review]
Instead of modifying every call, we could create a wrapper like this

File: apps/cornell/src/env.mjs [hunk 1]
```diff
@@ -15,6 +15,7 @@ const server = z.object({
 	DATABASE_URL: z.string(),
 	ADMINS: z.string().optional(),
 	STUDENTS: z.string().optional(),
+	ITELL_API_KEY: z.string().optional(),
```
Comment (qiushiyan) [inline apps/cornell/src/env.mjs:18]: Also need to add a new entry in the `processEnv` object below

**Comment by kyleburgess2025** [comment]
> Instead of modifying every call, we could create a factory like this
> 
> ```ts
> interface RequestOptions extends RequestInit {
>   bearerToken?: string;
> }
> 
> const createFetchWithBearerToken = (bearerToken?: string) => {
>   return async (url: string, options: RequestOptions = {}) => {
>     const { bearerToken: optionsBearerToken, ...requestOptions } = options;
>     const headers = new Headers(requestOptions.headers);
> 
> 
>     const token = optionsBearerToken || bearerToken;
>     if (token) {
>       headers.set('Authorization', `Bearer ${token}`);
>     }
> 
>     return fetch(url, {
>       ...requestOptions,
>       headers,
>     });
>   };
> };
> ```
> 
> Then create a itell specific fetch carrying the bearer token
> 
> ```ts
> const ifetch = createFetchWithBearerToken('api-token');
> ```
> 
> Then use `ifetch` to call the backend.
> 
> Could also get rid of the api routes and use `ifetch` directly in client components as well.

I don't think we can get rid of the API routes, we can't access the API key from the client components as the client should not have access to the secret key.

**Comment by qiushiyan** [comment]
You are right about using API routes. One thing is that `ifetch` only needs to be created once in one place, perhaps in a `lib/api.ts` file, the rest is looking good.

**Pull request merged**